### DISCUSSION TITLE
How do I create an array in Cpp2 which is on the heap instead of the stack?

### DISCUSSION THREAD

**Discussion by jituderaps** [hsutter/cppfront#743] [Q&A] [answered]
```
main: () = {
    p: std::unique_ptr<int> = new<int>[5](1, 2, 3, 4, 5);
}
```

**Comment by JohelEGP** [comment]
This happens to work (<https://cpp2.godbolt.org/z/8ehdn71W5>):
```Cpp2
main: () = {
  arr := new<int[]>(2u);
  [[assert: arr[0u] == 0]]
  [[assert: arr[1u] == 0]]
  [[assert: arr[2u] == 0]] // Uh oh...
}
```
Because <https://github.com/hsutter/cppfront/discussions/628#discussioncomment-6867808> and <https://github.com/hsutter/cppfront/discussions/628#discussioncomment-6867809> aren't implemented yet.
The last access is unchecked because `unique_ptr` for arrays erases the size.

**Reply by JohelEGP** [reply]
@hsutter For now, I think you might want the `new`'s `T` to not be an array type.

**Comment by AbhinavK00** [comment]
This works, somewhat
`a := new<std::array<T, N>>();`

it terminates on out of bounds access too.
Link : https://cpp2.godbolt.org/z/TKedT94a4

But idk how to initialise the elements

**Reply by JohelEGP** [reply]
Maybe as addressed at #729.

**Comment by hsutter** [comment, marked as answer]
See #740, and this is now addressed with commit e6a1cc363c54e5bcbf0381cd3fade6ef86c33f9d .

You can write: `pi := new<std::array<int, 5>>(1, 2, 3, 4, 5);`
